lib/pester.rb

```diff
@@ -56,12 +55,20 @@ def self.retry_action(opts = {}, &block)
         result = yield block
         return result
       rescue => e
-        class_reraise = opts[:retry_error_classes] && !opts[:retry_error_classes].include?(e.class)
+        retry_error_classes = opts[:retry_error_classes]
+        retry_error_messages = opts[:retry_error_messages]
+
+        class_retry = retry_error_classes && retry_error_classes.include?(e.class)
+        if class_retry && retry_error_messages && !retry_error_messages.any? { |m| e.message[m] }
+          class_retry = false
+        end
```

Comment:
i find this confusing -- maybe this more straightfoward?
      class_retry = retry_error_classes && retry_error_classes.include?(e.class)
      message_retry = retry_error_messages && !retry_error_messages.any? { |m| e.message[m] }
      retry = message_retry && class_retry